Automattic/jetpack
# External Media: Support new Google Photos Picker API

Fixes https://github.com/Automattic/jetpack/issues/40038

## Proposed changes:


* Hide the new flow behind the feature flag
* Update Google Photos screen supporting new Photos Picker API based on feature flag
  * If the flag doesn't exist, the flow remains the same
* Hide filtering since API doesn't support it anymore; filtering can be done in google's picker screen
<img width="1084" alt="Screenshot 2024-11-12 at 18 32 42" src="https://github.com/user-attachments/assets/049ea495-9e89-4f78-aa7c-db7221308217">

### Other information:


## Jetpack product discussion


- Project thread: pcmemI-3sx-p2
- Related PR: https://github.com/Automattic/wp-calypso/pull/95880
- Related PR: https://github.com/Automattic/jetpack/pull/40242

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Setup your sandbox and enable `keyring/includes/services/extended/google-photos.php` -> `is_picker_enabled` by adding your user id
* Go to the post editor, no matter whether simple or atomic site (follow [instructions](https://github.com/Automattic/jetpack/pull/40087#issuecomment-2462035498))
* Add the `/image` block and press "Select image"
* Select `Google Photos` and pick media from the connected account
* Once you are done with the Google Picker, select a media file and press the `Select` button
* Check if the changed flow looks and works fine
